engine/engine.go

```diff
@@ -358,8 +371,22 @@ func (q *Query) Explain() *ExplainOutputNode {
 	return explainVector(q.exec)
 }
 
-func (q *Query) Profile() {
-	// TODO(bwplotka): Return profile.
+func (q *Query) Analyze() *AnalyzeOutputNode {
+
+	return analyzeVector(q.obs)
+}
+func analyzeVector(obsv model.ObservableVectorOperator) *AnalyzeOutputNode {
+	ti, obs_vectors := obsv.Analyze()
```

Comment:
Let's not use snake case here, and name it a bit descriptively. 🙂 
```suggestion
	telemetry, obsVectors := obsv.Analyze()
```